tests/test_threading.py

```diff
@@ -0,0 +1,82 @@
+from threading import Thread
+import time
+import unittest
+
+import rasterio as rio
+from rasterio.env import get_gdal_config
+
+
+class TestThreading(unittest.TestCase):
```

Comment:
@geowurster Agreed. I got it in the next two commits: thread and process pool executors, with and without an `Env()` in the main thread.